components/sup/protocols/event.proto

```diff
@@ -72,6 +72,48 @@ message ServiceStoppedEvent {
   ServiceMetadata service_metadata = 2;
 }
 
+message ServiceUpdateStartedEvent {
+  EventMetadata event_metadata = 1;
+  ServiceMetadata service_metadata = 2;
+  // Details of the service pkg to which 
+  // we're attempting to update.These
+  // are implemented as strings for now,
+  // but at some point we'll likely want
+  // to create a type for this set of
+  // attributes (PackageIdent)
+  string update_origin = 3;
+  string update_name = 4;
+  string update_version = 5;
+  string update_release = 6;
```

Comment:
Is there a reason in Automate to have these as separate fields, rather than a single string?

Reply:
@cm i thought the ergonomics might be easier on the automate side instead of that code having to parse out what it needs for comparisons, etc. lmk if you disagree.

Reply:
@gpeers maybe we should chat about that usecase. It may be time to create some protobuf types to represent the different types of identifiers that we could have.

Reply:
@christophermaier i think i'll create a separate ticket and pr for any new types we decide on if that works for you.

Reply:
@gpeers I guess that depends on when Automate will start consuming this. If they're not going to do anything with this until those new types get added (and if the new types get added as the immediate next PR), then that's probably OK. I'd just hate for them to write code to deal with one form, only to have it changed right out from under them.

Alternatively, if it's not too much work, adding a new type to this PR would also be fine.

Reply:
@christophermaier ok, sure. let's discuss later today if you've got time.

Reply:
@baumanj appreciate the review : ) i don't believe so... these constitute the fully qualified identifier of the package we're updating to rather than the that of the current package. i've also refactored the code so we are consolidating the individual fields into one concatenated string which represents the FQI of the update package.

Reply:
The current and updating-to package must be the same in terms of `origin/name`, though, right?

Reply:
@baumanj yes, that's true. at this point in code we also have the version and release of the update package as well. are you ok with sending the update package ident to automate as a single string (origin/name/version/release) in the same way the current package ident is sent over?

Reply:
I'm torn. On one hand, I generally prefer to avoid mandatory duplication (especially in protocols) since it opens the door to inconsistent states. How should consumers behave if the `ServiceMetadata` says `foo/bar/XXX/YYY` but the update package says `baz/qux/WWW/ZZZ`? On the other, we don't have anywhere else where we have a release and version without origin and name, so it's a bit odd. I'd still say my preference would be populating `ServiceMetadata` with the pre-update FQPI and having just `update_version` and `update_release` in this message. If that leads to major awkwardness, I'll defer to your discretion.

Reply:
Hrmm... it's a good point.

Given that we don't (yet?) have a concept of just "version + release", I'm inclined to keep the code as it is now, sending the fully-qualified identifier of the new version.

As we dig into additional refactorings around our identifier types, though, this distinction could be good to keep in mind.

Reply:
We should be careful; changing protocols down the road is much harder than changing code. If we send FQPIs for both now, we're likely locked into that for awhile.